src/components/TodoForm/TodoForm.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './TodoForm.css';
+
+import { SelectOption } from '../SelectOption';
+
+export class TodoForm extends React.PureComponent {
+  state = {
+    todoTitle: '',
+    username: '',
+    todoUserId: 0,
+    todoId: this.props.todoId,
+    titleError: false,
+    usernameError: false,
+  }
+
+  addTodoUserId = (id) => {
+    this.setState({ todoUserId: Number(id) });
+  }
+
+  handleChange = (target) => {
+    this.setState({
+      username: target.value,
+      titleError: false,
+      usernameError: false,
+    });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    const { todoTitle, username } = this.state;
+
+    if (todoTitle === '' && username === '') {
+      this.setState({
+        usernameError: true,
+        titleError: true,
+      });
+
+      return;
+    }
+
+    if (todoTitle === '') {
+      this.setState({
+        titleError: true,
+      });
+
+      return;
+    }
+
+    if (username === '') {
+      this.setState({
+        usernameError: true,
+      });
+
+      return;
+    }
+
+    this.addTodo();
+
+    this.setState({
+      todoTitle: '',
+      username: '',
+      titleError: false,
+    });
+  }
+
+  addTodo() {
```

Comment:
let's move todo creating into the App component
addTodo will create new todo and add it to the state
and you will it in handleSubmit and pass everything you need